### PR TITLE
[Backport 1.3] Preserve URL Hash for SAML based login with Tests

### PR DESCRIPTION
**Pull request by scrawfor99**
Signed-off-by: Stephen Crawford <steecraw@amazon.com>

### Description
Manually backports SAML tests and updates from recent commits into 1.3. 

### Why these changes are required?
Resolves requests in #1219 discussion. 

### Check List
- [ ] ~New functionality includes testing~
- [ ] ~New functionality has been documented~
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
@scrawfor99 Could you look into the build/integration test failures?

**Comment by scrawfor99** [comment]
Update on the test failures: So looking at the errors, it seems that there is a versioning issue with the Cyprus tests that are called since the changes from the tests come from 2.4. I am looking through the different dependencies to try to find what needs to be switched. The error you get is ``` 

[2/4] Fetching packages...
error samlp@7.0.1: The engine "node" is incompatible with this module. Expected version ">=12". Got "10.24.1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
ERROR [bootstrap] failed:
ERROR Error: Command failed with exit code 1: /opt/hostedtoolcache/node/10.24.1/x64/lib/node_modules/yarn/bin/yarn.js install --non-interactive
          at makeError (/home/runner/work/security-dashboards-plugin/security-dashboards-plugin/OpenSearch-Dashboards/packages/osd-pm/dist/index.js:25139:11)
          at handlePromise (/home/runner/work/security-dashboards-plugin/security-dashboards-plugin/OpenSearch-Dashboards/packages/osd-pm/dist/index.js:24074:26)
          at process._tickCallback (internal/process/next_tick.js:68:7)
error Command failed with exit code 1.

``` 

During the run yarn step.

**Comment by scrawfor99** [comment]
Right now there is not a direct dependency on samlp. It is only on the saml-idp library which is dependent on a forked version of samlp: https://github.com/mcguinness/node-samlp/commits/master. I think we could maybe fork the saml-idp library and then swap the dependencies of it to the normal samlp and move it back (it is referencing an almost identical fork of samlp) but want to check and see what you think first.

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -88,7 +88,7 @@ jobs:
         run: |
           npm uninstall -g yarn
           echo "Installing yarn ${{ steps.versions_step.outputs.yarn_version }}"
-          npm i -g yarn@${{ steps.versions.outputs.yarn_version }}
+          npm i -g yarn@${{ steps.versions.outputs.yarn_version }} --ignore-engines
```
Comment (peternied) [inline .github/workflows/integration-test.yml:91]: Great idea! 🤞

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -2,19 +2,24 @@ name: Integration Tests
...
+  CI: 1
+
 jobs:
   tests:
     name: Run integration tests
     runs-on: ubuntu-latest
     steps:
       - name: Download OpenSearch Core
         run: |
-          wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/1.3.6/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-1.3.6-linux-x64.tar.gz
```
Comment (DarshitChanpura) [inline .github/workflows/integration-test.yml:12]: I believe these 2.4.0 version updates shouldn't be part of this PR

File: package.json [hunk 3]
```diff
@@ -15,20 +15,23 @@
     "build": "yarn plugin-helpers build && node build_tools/rename_zip.js",
     "start": "node ../../scripts/opensearch-dashboards --dev",
     "lint:es": "node ../../scripts/eslint",
-    "lint:sass": "node ../../scripts/sasslint",
-    "lint": "yarn run lint:es && yarn run lint:sass",
+    "lint": "yarn run lint:es && yarn run lint:style",
```
Comment (DarshitChanpura) [inline package.json:18]: Why remove the call to sasslint here?

File: public/apps/account/test/account-nav-button.test.tsx [hunk 4]
```diff
@@ -13,92 +13,121 @@
...
-  it('should set modal when show popup is true', () => {
-    (getShouldShowTenantPopup as jest.Mock).mockReturnValueOnce(true);
-    shallow(
-      <AccountNavButton
-        coreStart={mockCoreStart}
-        isInternalUser={true}
-        username={userName}
-        tenant="tenant1"
-        config={config as any}
+exports[`Account navigation button renders 1`] = `
```
Comment (DarshitChanpura) [inline public/apps/account/test/account-nav-button.test.tsx:16]: I'm not sure if you added this change. But this test file has been changed to a snapshot file. Do you know why is that?

File: .github/workflows/unit-test.yml [hunk 5]
```diff
@@ -13,7 +13,7 @@ jobs:
         with:
           path: OpenSearch-Dashboards
           repository: opensearch-project/OpenSearch-Dashboards
-          ref: '1.x'
+          ref: 'main'
```
Comment (DarshitChanpura) [inline .github/workflows/unit-test.yml:16]: why point to `main` here?

**Review by DarshitChanpura — COMMENTED** [review]
Left some review comments.

Also helpful to add `.idea` to `.gitignore` file so those changes are not pushed.
https://github.com/opensearch-project/security-dashboards-plugin/blob/3abb70774c42feecf9aa8202f959530be05bc2a3/.gitignore#L12

File: .github/workflows/integration-test.yml (hunk 2 above)
Comment (scrawfor99) [reply .github/workflows/integration-test.yml:12]: Yeah, I am just trying to get anything to work at this point. So I just copied the entire file to make sure that it was not a missing line somewhere that was causing the issue. I will be going back through to pair down the changes to those necessary once I get anything working.

**Comment by scrawfor99** [comment]
Closing in favor of #1226
